test/graphql/multi_target_search_test.go

```diff
@@ -208,3 +209,58 @@ func TestMultiTargetNearVector(t *testing.T) {
 		}
 	}
 }
+
+func TestMultiTargetNearVectorMultipleVectors(t *testing.T) {
+	err := testenv.SetupLocalWeaviate()
+	if err != nil {
+		t.Fatalf("failed to setup weaviate: %s", err)
+	}
+	client := testsuit.CreateTestClient(false)
+	ctx := context.TODO()
+
+	class := &models.Class{
+		Class: "MultiTargetNearVector",
+		VectorConfig: map[string]models.VectorConfig{
+			"first":  {Vectorizer: map[string]interface{}{"none": nil}, VectorIndexType: "hnsw"},
+			"second": {Vectorizer: map[string]interface{}{"none": nil}, VectorIndexType: "hnsw"},
+		},
+	}
+	client.Schema().ClassDeleter().WithClassName(class.Class).Do(ctx)
+	require.Nil(t, client.Schema().ClassCreator().WithClass(class).Do(ctx))
+	defer client.Schema().ClassDeleter().WithClassName(class.Class).Do(ctx)
+
+	objs := []*models.Object{
+		{Class: class.Class, Vectors: map[string]models.Vector{"first": []float32{1, 0, 0}, "second": []float32{1, 0, 0}}, ID: strfmt.UUID(uuid.New().String())},
+		{Class: class.Class, Vectors: map[string]models.Vector{"first": []float32{0, 0, 1}, "second": []float32{0, 0, 1}}, ID: strfmt.UUID(uuid.New().String())},
+	}
+
+	_, err = client.Batch().ObjectsBatcher().WithObjects(objs...).Do(ctx)
+	require.Nil(t, err)
```

Comment:
```suggestion
	require.NoError(t, err)
```

This would give a more readable message, like "Received unexpected error  'Whaam!'" instead of "expected nil, got &Error{...}"

Reply:
ohh, nice one!